ccm-connector/src/main/java/com/sequenceiq/cloudbreak/ccmimpl/termination/DefaultCcmResourceTerminationListener.java

```diff
@@ -0,0 +1,38 @@
+package com.sequenceiq.cloudbreak.ccmimpl.termination;
+
+import java.util.UUID;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.google.common.base.Throwables;
+import com.sequenceiq.cloudbreak.ccm.termination.CcmResourceTerminationListener;
+import com.sequenceiq.cloudbreak.ccmimpl.altus.GrpcMinaSshdManagementClient;
+
+/**
+ * Default CCM resource termination listener.
+ */
+@Component
+public class DefaultCcmResourceTerminationListener implements CcmResourceTerminationListener {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultCcmResourceTerminationListener.class);
+
+    @Inject
+    private GrpcMinaSshdManagementClient grpcMinaSshdManagementClient;
+
+    @Override
+    public void deregisterCcmSshTunnelingKey(@Nonnull String actorCrn, @Nonnull String accountId, @Nonnull String keyId) {
+        // JSA TODO get request ID from somewhere?
+        String requestId = UUID.randomUUID().toString();
```

Comment:
Redbeams and datalake have a `ThreadBasedRequestIdProvider` which gets the incoming request's request ID via a request filter. Maybe a similar mechanism is called for here.